apps/consume_analytics_metadata/src/main.rs

```diff
@@ -4,15 +4,15 @@
 use std::{ffi::CStr, process::abort, thread::sleep, time::Duration};
 
 use log::{error, info};
-use mdb::{Connection, Subscriber, SubscriberConfig};
+use mdb::{Connection, Error, Subscriber, SubscriberConfig};
 
 const TOPIC: &CStr = c"com.axis.analytics_scene_description.v0.beta";
 const SOURCE: &CStr = c"1";
 
 fn main() {
     acap_logging::init_logger();
 
-    let connection = Connection::try_new(Some(|e| {
+    let connection = Connection::try_new(Some(|e: &Error| {
```

Comment:
When I test I see this is necessary but I don't understand why, so if you know please share.

Reply:
I don't exactly know either, the compiler thinks the closure might capture the environment which limits the lifetime on `e` which must by any lifetime. Making it non-optional works though, is there a particular reason this callback is optional but all the other ones arent?

Reply:
I think the underlying C API works that way. If I were to guess why I'd say it is plausible that we want to create a connection without a callback telling us how it went, but it makes no sense to subscribe to messages without a method of handling the received messages.

Reply:
Fair enough